onnx/defs/tensor/defs.cc

```diff
@@ -53,10 +53,11 @@ from the shape argument.)DOC")
 	            "Constrain input and output types to float tensors.");
 
 OPERATOR_SCHEMA(Concat)
+.SinceVersion(3)
 .Attr("axis",
     "Which axis to concat on",
     AttributeProto::INT,
-    OPTIONAL)
+    REQUIRED)
```

Comment:
"required" is the default value, so you may just remove "OPTIONAL" here. Ideally, there're only two APIs, one for required attribute, one for optional attribute with default value (and remove this flag finally).